tvtk/special_gen.py

```diff
@@ -319,8 +319,11 @@ def __iter__(self):
                 yield obj.GetPoint(i)
 
         def _check_key(self, key, n):
-            if type(key) != type(1):
-                raise TypeError, "Only integers are valid keys."
+            ##############################################
+            # Allow int and long keys. Fixes GH Issue 173.
+            ##############################################
+            if type(key) not in [type(1), type(1L)]:
```

Comment:
IIRC, the main reason for `_check_key` is to support negative indices and raise the correct exceptions where necessary.  VTK will not raise any exceptions for those cases, which could lead to a segfault.  Try this:  
```
import vtk
d = vtk.vtkDoubleArray()
d.SetNumberOfTuples(5)
d.GetTuple(-1)
d.GetTuple(100)
```
You might get a segfault!  The integer/long checks are not technically necessary but should not be harmful.  VTK will raise the exception but the user may be confused when they see the exception so I think I caught it earlier.